### PR TITLE
Improve Tesla docs

### PR DESCRIPTION
**Pull request by WLSF**
* Small improvements for the module `Tesla` options description
* Add example on how to use `:except` options
* Add example on how to use build_url with Keyword list. 
* Add tests for `build_url/2`

### CODE REVIEW

File: lib/tesla.ex [hunk 1]
```diff
@@ -490,6 +503,23 @@ defmodule Tesla do
...
   def build_adapter(fun), do: Tesla.Builder.client([], [], fun)
 
+  @doc """
+  Builds URL with the given query params.
+
+  Useful when you need to create an URL with dynamic query params from a Keyword list
+
+  ## Example
+
+      iex> Tesla.build_url("http://api.example.com", [{:user, 3}, {:page, 2}])
```
Comment (teamon) [inline lib/tesla.ex:513]:
```suggestion
      iex> Tesla.build_url("http://api.example.com", [user: 3, page: 2])
```

File: lib/tesla.ex [hunk 2]
```diff
@@ -490,6 +503,23 @@ defmodule Tesla do
...
+      "http://api.example.com?user=3&page=2"
+
+      # URL that already contains query params
+      iex> url = "http://api.example.com?user=3"
+      iex> Tesla.build_url(url, [{:page, 2}, {:status, true}])
```
Comment (teamon) [inline lib/tesla.ex:518]:
```suggestion
      iex> Tesla.build_url(url, [page: 2, status: true])
```

File: test/tesla_test.exs [hunk 3]
```diff
@@ -273,4 +273,34 @@ defmodule TeslaTest do
...
     end
   end
+
+  describe "build_url/2" do
+    setup do
+      {:ok, url: "http://api.example.com"}
+    end
+
+    test "returns URL with query params from keyword list", %{url: url} do
+      query_params = [{:user, 3}, {:page, 2}]
```
Comment (teamon) [inline test/tesla_test.exs:283]:
```suggestion
      query_params = [user: 3, page: 2]
```

File: test/tesla_test.exs [hunk 4]
```diff
@@ -273,4 +273,34 @@ defmodule TeslaTest do
...
+      assert build_url(url, query_params) === url <> "?user=3&page=2"
+    end
+
+    test "returns URL with new query params concated from keyword list", %{url: url} do
+      url_with_param = url <> "?user=4"
+      query_params = [{:page, 2}, {:status, true}]
```
Comment (teamon) [inline test/tesla_test.exs:289]:
```suggestion
      query_params = [page: 2, status: true]
```

Comment (WLSF) [reply test/tesla_test.exs]: Wow, I didn't realize that having a list of tuples could be the same as having a list of keywords, thanks!

**Comment by WLSF** [comment]
@teamon thanks for the suggestions!

**Pull request merged**